ModularTegustation/tegu_items/lc13_uniqueitems.dm

```diff
@@ -0,0 +1,559 @@
+	..() //also
+	if(ishuman(user))
+		user.apply_status_effect(STATUS_EFFECT_SANITYINSURANCE)
+
+/obj/item/moonlightstone/dropped(mob/living/carbon/human/user)
+	..()
+	user.remove_status_effect(STATUS_EFFECT_SANITYINSURANCE)
+
+/datum/status_effect/sanity_insurance
+	id = "sanity insurance"
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/sanity_insurance
+
+/datum/status_effect/sanity_insurance/tick() //i dont think there is any way to avoid using tick() Nevermind the managerbullet can use signals that are given out by certain events.
+	. = ..()
+	var/mob/living/carbon/human/H = owner
+	if(H.stat == DEAD)
+		for(var/obj/item/moonlightstone/O in owner.GetAllContents())
+			qdel(O)
+		qdel(src)
+	if(H.sanity_lost) //|| means 'or', && means 'and'
+		H.adjustSanityLoss(H.maxSanity+1)
+		H.adjustSanityLoss(H.maxSanity*(-0.95))
+		playsound(get_turf(H), 'sound/effects/glassbr1.ogg', 50, 0, 4)
+		for(var/obj/item/moonlightstone/O in owner.GetAllContents())
+			qdel(O)
+		qdel(src)
+
+/atom/movable/screen/alert/status_effect/sanity_insurance //experimental and used to inform the owner that they have sanity insurance.
+	name = "sanity insurance"
+	icon = 'ModularTegustation/Teguicons/lc13icons.dmi'
+	desc = "To bring you back when your sanity fails you."
+	icon_state = "prudence_icon"
+
+#undef STATUS_EFFECT_SANITYINSURANCE
+
+	//Defective Manager Bullet PLACEHOLDER OR PROTOTYPE SHIELDS
+/obj/item/managerbullet
+	name = "prototype manager bullet"
+	desc = "Welfare put out a notice that they lost a bunch of manager bullets. This must be one of them."
+	icon = 'icons/obj/ammo.dmi'
+	icon_state = "sleeper-live"
+
+/obj/item/managerbullet/attack_self(mob/living/carbon/user) //shields from basegame are Physical Intervention Shield, Trauma Shield, Erosion Shield, Pale Shield
+	user.visible_message("<span class='notice'>[user] smashes [src] against their chest.</span>")
+	playsound(get_turf(user), 'sound/effects/pop_expl.ogg', 5, 0, 3)
+	var/randomnum = rand(1,3)
+	switch(randomnum)
+		if(1)
+			user.apply_status_effect(/datum/status_effect/interventionshield)
+		if(2)
+			user.apply_status_effect(/datum/status_effect/interventionshield/white)
+		if(3)
+			user.apply_status_effect(/datum/status_effect/interventionshield/black)
+	qdel(src)
+
+/datum/status_effect/interventionshield
+	id = "physical intervention shield"
+	duration = 15 SECONDS
+	status_type = STATUS_EFFECT_REPLACE
+	alert_type = null
+	var/inherentarmorcheck
+	var/statuseffectvisual = icon('ModularTegustation/Teguicons/tegu_effects.dmi', "red_shield")
+	var/shieldhealth = 100
+	var/damagetaken = 0
+	var/respectivedamage = RED_DAMAGE
+	var/faltering = 0
+
+/datum/status_effect/interventionshield/on_apply()
+	. = ..()
+	owner.add_overlay(statuseffectvisual)
+	var/mob/living/carbon/human/L = owner
+	switch(respectivedamage)
+		if(RED_DAMAGE)
+			inherentarmorcheck = L.physiology.red_mod
+			L.physiology.red_mod *= 0.0001
```

Comment:
This is certainly not how we should do it...

Reply:
Should i remove it? From my tests it seems to work and fear damage bypasses it completely

Reply:
This is also how life for a daredevil also handles it.